src/main/java/de/tum/in/www1/artemis/web/rest/PlagiarismResource.java

```diff
@@ -187,6 +187,33 @@ else if (comparison.getSubmissionB().getStudentLogin().equals(studentLogin)) {
         return ResponseEntity.ok(new PlagiarismCaseDTO(anonymizedComparisonForStudentView.getPlagiarismResult().getExercise(), Set.of(anonymizedComparisonForStudentView)));
     }
 
+    /**
+     * Retrieves the plagiarismComparison specified by its Id.
+     *
+     * @param courseId the id of the course
+     * @param comparisonId the id of the PlagiarismComparison
+     * @return the PlagiarismComparison
+     * @throws AccessForbiddenException if the requesting user is not affected by the plagiarism case.
+     */
+    @GetMapping("courses/{courseId}/plagiarism-comparisons/{comparisonId}/for-split-view")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<PlagiarismComparison<?>> getPlagiarismComparisonForEditor(@PathVariable("courseId") long courseId, @PathVariable("comparisonId") Long comparisonId) {
+        var comparisonA = plagiarismComparisonRepository.findByIdWithSubmissionsStudentsAndElementsAElseThrow(comparisonId);
```

Comment:
I guess this could be the reason for the missing anonymisation. You are retrieving the comparisons without anonymizing them.